workers/reporter/src/main/kotlin/reporter/CustomLicenseTextProvider.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.workers.reporter
+
+import org.eclipse.apoapsis.ortserver.config.Path
+import org.eclipse.apoapsis.ortserver.workers.common.context.WorkerContext
+import org.eclipse.apoapsis.ortserver.workers.common.resolvedConfigurationContext
+
+import org.ossreviewtoolkit.reporter.DefaultLicenseTextProvider
+import org.ossreviewtoolkit.reporter.LicenseTextProvider
+
+import org.slf4j.LoggerFactory
+
+private val logger = LoggerFactory.getLogger(CustomLicenseTextProvider::class.java)
+
+/**
+ * A specialized [LicenseTextProvider] implementation that is used by the [ReporterRunner] to support efficient access
+ * to custom license texts stored in a configuration directory.
+ *
+ * The [LicenseTextProvider] implementations available in ORT expect that license texts are available in a local
+ * folder. For ORT Server, this is not necessarily the case, but depends on the provider for config files. This
+ * implementation therefore uses the configuration manager API to check whether a requested license text is available
+ * in the configuration. If it is, it obtains the license text from the configuration. Otherwise, it delegates to
+ * another [LicenseTextProvider] implementation which searches the default paths used by ORT.
+ */
+internal class CustomLicenseTextProvider(
+    /** The current worker context. */
+    val workerContext: WorkerContext,
```

Comment:
`workerContext` has lots of properties but this class needs only the `configManager` and the `resolvedConfigurationContext`. Passing those properties here instead of the `workerContext` would reduce dependencies of the class and simplify the test.